# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/wifi/dbus/DbusWifiScanner.h
# Commit: afd62e82fe83
# Full Hash: afd62e82fe8320776e9696e3a8881eaae56b8822
# Author: David Parks <daparks@mozilla.com>
# Date: 2023-05-11 04:06:39
# Regressor Bug: 1810421
# File Overlap Count: 2
# Description:
#   Bug 1810421: Refactor wifi scanning to consolidate generic code r=necko-reviewers,kershaw,valentin
#   
#   Reorganizes the wifi-scanning code and makes more of it platform-generic
#   to ease the transition from polling the wifi to usually scanning only on
#   network changes.  This is essentially just moving files/code around and
# ==============================================================================

diff -r 9c94ab8b34c0 -r afd62e82fe83 netwerk/wifi/dbus/DbusWifiScanner.h
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/netwerk/wifi/dbus/DbusWifiScanner.h	Wed May 10 19:46:29 2023 +0000
@@ -0,0 +1,60 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef NSWIFIAPSCANNERDBUS_H_
+#define NSWIFIAPSCANNERDBUS_H_
+
+#define DBUS_API_SUBJECT_TO_CHANGE
+#include <dbus/dbus.h>
+#include <dbus/dbus-glib-lowlevel.h>
+
+#include "WifiScanner.h"
+
+class nsIWifiAccessPoint;
+class nsWifiAccessPoint;
+
+namespace mozilla {
+
+using AccessPointArray = nsTArray<RefPtr<nsIWifiAccessPoint>>;
+
+class WifiScannerImpl final : public WifiScanner {
+ public:
+  explicit WifiScannerImpl();
+  ~WifiScannerImpl();
+
+  /**
+   * GetAccessPointsFromWLAN
+   *
+   * Scans the available wireless interfaces for nearby access points and
+   * populates the supplied collection with them
+   *
+   * @param accessPoints The collection to populate with available APs
+   * @return NS_OK on success, failure codes on failure
+   */
+  nsresult GetAccessPointsFromWLAN(AccessPointArray& accessPoints);
+
+ private:
+  nsresult SendGetDevices(AccessPointArray& accessPoints);
+  nsresult SendGetDeviceType(const char* aPath, AccessPointArray& accessPoints);
+  nsresult SendGetAccessPoints(const char* aPath,
+                               AccessPointArray& accessPoints);
+  nsresult SendGetAPProperties(const char* aPath,
+                               AccessPointArray& accessPoints);
+  nsresult IdentifyDevices(DBusMessage* aMsg, AccessPointArray& accessPoints);
+  nsresult IdentifyDeviceType(DBusMessage* aMsg, const char* aDevicePath,
+                              AccessPointArray& accessPoints);
+  nsresult IdentifyAccessPoints(DBusMessage* aMsg,
+                                AccessPointArray& accessPoints);
+  nsresult IdentifyAPProperties(DBusMessage* aMsg,
+                                AccessPointArray& accessPoints);
+  nsresult StoreSsid(DBusMessageIter* aVariant, nsWifiAccessPoint* aAp);
+  nsresult SetMac(DBusMessageIter* aVariant, nsWifiAccessPoint* aAp);
+  nsresult GetDBusIterator(DBusMessage* aMsg, DBusMessageIter* aIterArray);
+
+  RefPtr<DBusConnection> mConnection;
+};
+
+}  // namespace mozilla
+
+#endif  // NSWIFIAPSCANNERDBUS_H_